src/components/Discover/DiscoverSearchInput.tsx

```diff
@@ -122,17 +119,25 @@ type DiscoverSearchInputProps = {
 
 const DiscoverSearchInput = ({
   isDiscover,
-  isLoading = false,
-  onChangeText,
   onFocus,
   onBlur,
-  searchQuery,
   testID,
   placeholderText = lang.t('button.exchange_search_placeholder'),
   clearTextOnFocus = true,
   currentChainId,
 }: DiscoverSearchInputProps) => {
-  const { searchInputRef, isSearching } = useDiscoverScreenContext();
+  const { isSearching, searchQuery } = useDiscoverSearchQueryStore(state => {
+    return {
+      searchQuery: state.searchQuery,
+      isSearching: state.isSearching,
+    };
+  });
```

Comment:
```suggestion
  const isSearching = useDiscoverSearchQueryStore(state => state.isSearching);
  const searchQuery = useDiscoverSearchQueryStore(state => state.searchQuery);
```
iirc we can potentially drop a few rerenders if we select the primitives instead of an object